pkgs/top-level/splice.nix

```diff
@@ -37,15 +37,18 @@ let
       inherit name;
       value = let
         defaultValue = mash.${name};
-        buildValue = buildPkgs.${name} or {};
-        runValue = runPkgs.${name} or {};
+        buildValue = buildPkgs.${name};
+        runValue = runPkgs.${name};
         augmentedValue = defaultValue
           // (lib.optionalAttrs (buildPkgs ? ${name}) { nativeDrv = buildValue; })
           // (lib.optionalAttrs (runPkgs ? ${name}) { crossDrv = runValue; });
         # Get the set of outputs of a derivation
-        getOutputs = value: lib.genAttrs
-          (value.outputs or (lib.optional (value ? out) "out"))
-          (output: value.${output});
+        getOutputs = value0: let
+            eval = builtins.tryEval value0;
+            value = if eval.success then eval.value else {};
+          in lib.genAttrs
```

Comment:
This indentation does not follow rules that can reasonably be encoded in a pretty printer: either put the `let` above on a new line, or dedent the 5 lines below it by 2 spaces.